src/latest-release.ts

```diff
@@ -0,0 +1,31 @@
+import * as nodegit from "nodegit";
+import * as semver from "semver";
+
+
+export async function checkoutLatestRelease(
+    repo: nodegit.Repository,
+    dir?: string)
+        : Promise<nodegit.Repository> {
+  const tags = await nodegit.Tag.list(repo);
+  let latestRelease: string;
+  if (tags && tags.length > 0) {
+    try {
+      latestRelease = tags.sort(semver.rcompare)[0];
+    } catch (err) {
+      // We couldn't pick a release, so we'll just bow out here.
+      return repo;
+    }
+  } else {
+    return repo;
+  }
+  let commit: nodegit.Commit;
+  try {
+    commit = await repo.getReferenceCommit(latestRelease);
```

Comment:
what I was getting at with my last comment about why `getReferenceCommit` throws: it seems like this function shouldn't catch the error, but cause the whole program to fail. I can't imagine a case where you got a release tag and then couldn't use it that wasn't an unexpected failure. The user probably wants to know when they aren't using the latest release like they are expecting.